backend/services/memory/project_detail_service.py

```
@@ -49,14 +49,22 @@ async def ingest_project_detail(self, project_id: str, raw_text: str, mode: str
+            final_text = await llm_service.chat_with_system_prompt(
                 "Perform sophisticated merge of existing project detail with new insights", 
                 f"{merge_system_prompt}\n\n{merge_input}"
             )
         else:
             # For replace mode or when no existing content, use simpler synthesis
             synthesis_prompt = self._build_synthesis_system_prompt()
-            final_text = await self.gemini.chat_with_system_prompt(
+            llm_service = self.llm_provider_service.get_llm_service_by_project_id(project_id)
+            if llm_service is None:
+                logger.error(f"No LLM service available for project {project_id}")
+                raise ValueError(f"No LLM service available for project {project_id}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)